source/extensions/quic_listeners/quiche/BUILD

```diff
@@ -356,6 +356,7 @@ envoy_cc_library(
     srcs = ["quic_transport_socket_factory.cc"],
     hdrs = ["quic_transport_socket_factory.h"],
     tags = ["nofips"],
+    visibility = ["//visibility:public"],
```

Comment:
given that we discourage extensions with public visibility can you add a comment?

you could use extra_visibility to scope it to source/server and comment that the factory code is needed for config validation.